filesystem/apm_partition_table.ksy

```diff
@@ -0,0 +1,69 @@
+meta:
+  id: apm_partition_table
+  title: APM (Apple Partition Map) partition table
+  license: CC0-1.0
+  endian: be
+  encoding: ascii
+doc-ref: Specification taken from https://en.wikipedia.org/wiki/Apple_Partition_Map
+instances:
+  sector_size:
+    value: 0x200
+    # APM only supports 0x200 (512) byte sectors.
+  partition_lookup:
+    io: _root._io
+    pos: _root.sector_size
+    type: partition_entry
+    size: sector_size
+    # Every partition entry contains the number of partition entries.
+    # We parse the first entry, to know how many to parse, including the first one.
+    # No logic is given what to do if other entries have a different number.
```

Comment:
Technically, you can just do something like to look that up without parsing everything:

```yaml
instances:
  num_partitions:
    pos: sector_size + 4
    type: u4
```

I'm not sure it's any better, though...

Reply:
Well, if Apple is using a regular partition entry as the header, then I would say, reuse the correct logic.